File: apps/api/internal/transport/http/responses/order.go
package responses

import (
	"time"

	"github.com/google/uuid"
)

type OrderDecision struct {
	ContentNote IContentNote `json:"contentNote,omitempty"`
	DecidedAt   time.Time    `json:"decidedAt"`
	DecidedBy   *User        `json:"decidedBy"`
	Status      string       `json:"status"`
}

type Order struct {
	ID        uuid.UUID `json:"id"`
	CreatedAt time.Time `json:"createdAt"`
	Source    string    `json:"source"`
	Anonymous bool      `json:"anonymous"`
	Orderer   *Orderer  `json:"orderer,omitempty"`
}

type DetailedOrder struct {
	*Order
	Category string   `json:"category"`
	Content  IContent `json:"content,omitempty"`
	Message  string   `json:"message"`
}

type OrderWithDecision struct {
	*DetailedOrder
	Decision *OrderDecision `json:"decision,omitempty"`
}

File: apps/web/lib/model/order.ts
import { ContentCategory } from './content'
import { ContentNote } from './content-note'
import { Orderer } from './orderer'
import { User } from './user'

export type Order = {
  id: string
  createdAt: string
  source: string
  anonymous: boolean
  orderer?: Orderer
}

export type DetailedOrder = Order & {
  category: ContentCategory
  message: string
}

export type OrderWithDecision = DetailedOrder & {
  decision?: OrderDecision
}

export type OrderDecisionStatus = 'approved' | 'rejected'

export type OrderDecision = {
  contentNote?: ContentNote
  decidedAt: string
  decidedBy?: User
  status: OrderDecisionStatus
}
